src/generators/app.ts

```diff
@@ -233,10 +239,12 @@ class App extends Generator {
     this.pjson.repository = this.answers.github ? `${this.answers.github.user}/${this.answers.github.repo}` : defaults.repository
     this.pjson.scripts.posttest = 'yarn run lint'
     // this.pjson.scripts.precommit = 'yarn run lint'
-    if (this.ts && this.mocha) {
+    if (this.ts && this.tslint && this.mocha) {
       this.pjson.scripts.lint = 'tsc -p test --noEmit && tslint -p test -t stylish'
-    } else if (this.ts) {
+    } else if (this.ts && this.tslint) {
       this.pjson.scripts.lint = 'tsc -p . --noEmit && tslint -p . -t stylish'
+    } else if (this.ts && this.mocha) {
+      this.pjson.scripts.lint = 'tsc -p test --noEmit'
```

Comment:
hmm, we use `-p test` when use mocha and `-p .` when we don't have mocha (this way it builds the mocha files only when doing the linting)

Maybe we could rewrite this like the following to clean it up:

```js
if (this.ts) {
  const tsProject = this.mocha ? 'test' : '.'
  this.pjson.scripts.lint = `tsc -p ${tsProject}`
  if (this.tslint) this.pjson.scripts.lint += `&& tslint -p ${tsProject}`
}
```

I _think_ that's right. What do you think?